outlines/integrations/llamacpp.py

```diff
@@ -66,6 +66,16 @@ def __init__(self, model: "Llama"):
     def convert_token_to_string(self, token: str) -> str:
         return token
 
+    def __getstate__(self):
+        """Allow tokenizer to be used as hash key by excluding self.decode"""
+        return (
+            sorted(self.vocabulary.items()),
```

Comment:
Is `sorted` necessary here?

Reply:
I don't think so. If we use a `llama_cpp` tokenizer we decode sequentially by token ID. HF tokenizers `get_vocab()` has consistent ordering. I experimented with both, and they were fine without `sorted()`.

Fixed.